components/script/dom/globalscope.rs

```diff
@@ -3211,8 +3213,26 @@ impl GlobalScope {
                                 }
                             };
 
-                            // TODO: 6. If navigator.[[hasGamepadGesture]] is false
-                            //          and gamepad contains a gamepad user gesture:
+                            if !window.Navigator().has_gamepad_gesture() {
+                                window.Navigator().set_has_gamepad_gesture(true);
```

Comment:
We should only set it to true according to https://w3c.github.io/gamepad/#dfn-gamepad-user-gesture. Is this dealt with at the embedding layer? It might relate to what you experienced initially as the pad not responding to a first input.

Reply:
I suppose it's a bit of a simplification as it is now, I'm treating any input as a gamepad user gesture but I should technically be making it more rigorous and checking for axis threshold too. This is unrelated to my initial issue with it not responding to first input though, that was caused by GilRs not emitting a changed event when a button doesn't have prior state (e.g. pressed for the very first time).